.github/workflows/tests.yml

```diff
@@ -0,0 +1,33 @@
+name: RapiDAST
+
+on:
+  push:
+    branches: ["development", "main"]
+  pull_request:
+    branches: ["development", "main"]
+
+permissions:
+  contents: read
```

Comment:
Do you know how the workflow will behave if any of the pre-commit tasks tries to make changes to the code, for example the `reorder-python-imports` ?

Reply:
It would have modified the copy of the code at the linting step and then I believe tested the modified code in the subsequent step. I think this is undesirable so I've swapped the order of steps, code is tested first, then linted (which can end up in modified code).

Currently all the steps run sequentially, though some could be made to run in parallel I feel this is good enough for now. I will parallelize these steps when we move to a tekton based model.

I believe there's also work planned to use ruff for linting, which could replace many of the pre-commit linters and should be easier to make only check the code, not modify it. Til then this approach is low cost to centralize and automate the established tests/linters.